core/src/main/java/org/elasticsearch/search/aggregations/AggregatorBase.java

```diff
@@ -96,6 +102,14 @@ public boolean needsScores() {
                 return false; // unreachable
             }
         };
+        try {
+            this.breakerService
+                    .getBreaker(CircuitBreaker.BUCKET)
+                    .addEstimateBytesAndMaybeBreak(1024, "<agg [" + name + "]>");
+        } catch (CircuitBreakingException cbe) {
+            this.failed = true;
+            throw cbe;
+        }
```

Comment:
This part I like since it is growing based on the number of aggregator instances, which is not accounted today.